.clang-tidy

```diff
@@ -1,3 +1,192 @@
+# Some checks are suppressed:
+#
+# This check is useless for us. Many objects (like tensors or problem descriptions)
+# have mutiple parameters of the same type
+#   -bugprone-easily-swappable-parameters
+#
+# Too many narrowing conversions in our code
+#   -bugprone-narrowing-conversions
+#
+# We shouldn't be using rand()
+#   -cert-msc30-c
+#
+# We really shouldn't use bitwise operators with signed integers, but opencl leaves us no choice
+#   -hicpp-signed-bitwise
+#
+# This one is extremely slow, and probably has lots of FPs
+#   -misc-confusable-identifiers
+#
+# TODO We are not ready to use it, but very useful
+#   -readability-function-cognitive-complexity
+#
+# We dont think this is a useful check. Disabled on migraphx
+#   -readability-identifier-length
+#
+# There are many FPs with this, let's disable it (ditto in MIGraphX)
+#   -readability-suspicious-call-argument
+#
+# TODO Code Quality WORKAROUND ROCm 5.1 update
+#   -cert-err33-c
+#   -google-readability-casting
+#   -hicpp-use-emplace
+#   -modernize-use-emplace
+#   -performance-unnecessary-copy-initialization
+#   -readability-container-data-pointer
+#
+# TODO Code Quality WORKAROUND ROCm 5.3 && Ubuntu 22.04 && C++17 && cppcheck 2.9 update
+#   -bugprone-use-after-move
```

Comment:
[Notice] We are loosing some flexibility. For example, the comments about tidy checks cannot be placed nearby the checks anymore